### PR TITLE
Sklearn estimator

### PR DESCRIPTION
**Pull request by granawkins**

### CODE REVIEW

File: karoo_gp/test/test_sklearn.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from sklearn.utils.estimator_checks import check_estimator
+from karoo_gp import BaseGP
+
+def test_sklearn_estimator():
+    # check_estimator(BaseGP())
+    # TODO: failing on parse_node:
+    # "ValueError: Integers to negative integer powers are not allowed."
+    pass
```
Comment (ezio-melotti) [inline karoo_gp/test/test_sklearn.py:8]: A skip would be better.

File: karoo_gp/test/test_tree.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import json
 
 from karoo_gp import Tree, Branch
+from .util import dump_json
```
Comment (ezio-melotti) [inline karoo_gp/test/test_tree.py:6]: This doesn't seem to be used.

File: karoo-gp.py [hunk 3]
```diff
@@ -583,7 +583,7 @@ def fx_karoo_pause(model):
     precision=precision,
     swim=swim,
     mode=mode,
-    seed=seed,
+    random_state=seed,
```
[the remark this answers is not in the document]
Comment (granawkins) [reply karoo-gp.py:586]: Ya it's an `sklearn` convention, see [here](https://scikit-learn.org/stable/developers/develop.html#random-numbers). We're also using their `utils.check_random_state` to validate the user's input. It doesn't seem to break anything, and given the lengths we're already going to conform with sklearn, might as well.

File: karoo_gp/test/test_tree.py (hunk 2 above)
Comment (granawkins) [reply karoo_gp/test/test_tree.py:6]: It's used on line 92 but normally commented out. Lets you quickly update the tests to check what's wrong or update the baseline.

File: karoo_gp/test/test_sklearn.py (hunk 1 above)
Comment (granawkins) [reply karoo_gp/test/test_sklearn.py:8]: What do you mean?

Comment (ezio-melotti) [reply karoo_gp/test/test_sklearn.py:8]: https://docs.pytest.org/en/7.1.x/how-to/skipping.html#skipping-test-functions

Comment (granawkins) [reply karoo_gp/test/test_sklearn.py:8]: Ah roger that

**Pull request merged**